app/main.py

```diff
@@ -1,25 +1,99 @@
+from __future__ import annotations
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+    def __init__(
+            self,
+            min_amount: int,
+            max_amount: int
+    ) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(
+            self,
+            owner: SlideLimitationValidator,
+            name: str
+    ) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(
+            self,
+            instance: SlideLimitationValidator,
+            owner: SlideLimitationValidator
+    ) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(
+            self,
+            instance: SlideLimitationValidator,
+            value: int
+    ) -> None:
+        if not self.min_amount <= value <= self.max_amount:
+            setattr(instance, self.protected_name, False)
+            return
```

Comment:
In the `__set__` method, you are setting the attribute to `False` if the value does not meet the condition. This might be confusing because the attribute can be an integer or a boolean. It would be more appropriate to raise an exception.